### PR TITLE
Update func_cut.c

### PR DESCRIPTION
**Pull request by phoneben**
Func. CUT add note of unusual syntax for func. 

want to update the WIKI
The **varname**  parameter must be a variable name, not a variable expression
`exten => s,1,Set(foo=${CUT(bar,,2)}) ; This is correct syntax`
`exten => s,1,Set(foo=${CUT(${bar},,2)}) ; This is invalid syntax (unless bar contains the name of another variable)`

### CODE REVIEW

**Comment by phoneben** [comment]
cherry-pick-to: 20
cherry-pick-to: 18

**Review by InterLinked1 — CHANGES_REQUESTED** [review]
The change should be squashed into a single commit.
Additionally your commit message doesn't meet the guidelines: https://docs.asterisk.org/Development/Policies-and-Procedures/Commit-Messages/

File: funcs/func_cut.c [hunk 1]
```diff
@@ -72,10 +72,14 @@
 				<para>Number of the field you want (1-based offset), may also be specified as a range (with <literal>-</literal>)
 				or group of ranges and fields (with <literal>&amp;</literal>)</para>
 			</parameter>
+ 					<example title="The varname parameter must be a variable name, not a string value. This is unusual syntax. So:">
+				exten => s,1,Set(foo=${CUT(bar,,2)}) ; This is correct syntax
+				exten => s,1,Set(foo=${CUT(${bar},,2)}) ; This is invalid syntax (unless bar contains the name of another variable)
```
Comment (InterLinked1) [inline funcs/func_cut.c:77]: Maybe use priority 2, since this isn't actually valid dialplan otherwise?

File: funcs/func_cut.c [hunk 2]
```diff
@@ -72,10 +72,14 @@
 				<para>Number of the field you want (1-based offset), may also be specified as a range (with <literal>-</literal>)
 				or group of ranges and fields (with <literal>&amp;</literal>)</para>
 			</parameter>
+ 					<example title="The varname parameter must be a variable name, not a string value. This is unusual syntax. So:">
```
Comment (InterLinked1) [inline funcs/func_cut.c:75]: "Unusual" seems kind of normative to me... there are several dialplan functions that work similarly. Also example titles should be relatively short. Also it's a bit confusing as currently worded... variable names are string values. I think it'd be fine to rephrase to "The varname parameter must be a variable name, not a variable expression". A colon is not needed at the end.

**Review by InterLinked1 — CHANGES_REQUESTED** [review]
The merge commit just made the problem worse... now there are 4 separate commits, up from 3.

File: funcs/func_cut.c [hunk 3]
```diff
@@ -72,10 +72,14 @@
...
+				exten => s,1,Set(foo=${CUT(bar,,2)}) ; This is correct syntax
+				exten => s,1,Set(foo=${CUT(${bar},,2)}) ; This is invalid syntax (unless bar contains the name of another variable)		
+					</example>
 		</syntax>
 		<description>
 			<para>Cut out information from a string (<replaceable>varname</replaceable>), based upon a named delimiter.</para>
-		</description>
+		</description>
```
[the remark this answers is not in the document]
Comment (phoneben) [reply funcs/func_cut.c]: done

File: funcs/func_cut.c (hunk 2 above)
Comment (phoneben) [reply funcs/func_cut.c]: fix you wording ty ,
as for the usual in the dialpan most of the function use `${} `this one is different
`Set(foo=${CUT(bar,,${counter})})`
bar is whithout and counter would need to be with the `${} `

File: funcs/func_cut.c [hunk 4]
```diff
@@ -71,10 +71,14 @@
 			<parameter name="range-spec" required="true">
 				<para>Number of the field you want (1-based offset), may also be specified as a range (with <literal>-</literal>)
 				or group of ranges and fields (with <literal>&amp;</literal>)</para>
-			</parameter>
+   			</parameter>
```
Comment (InterLinked1) [inline funcs/func_cut.c:74]: stray leading whitespace

File: funcs/func_cut.c [hunk 5]
```diff
@@ -71,10 +71,14 @@
...
 		</syntax>
 		<description>
 			<para>Cut out information from a string (<replaceable>varname</replaceable>), based upon a named delimiter.</para>
+  					 <example title="the varname parameter must be a variable name, not a variable expression">
```
Comment (InterLinked1) [inline funcs/func_cut.c:78]: Capitalize the beginning of the title (the)

**Review by InterLinked1 — CHANGES_REQUESTED** [review]

**Review by gtjoseph — CHANGES_REQUESTED** [review]
All the commits need to be squashed into one.

**Comment by gtjoseph** [comment]
Oh, and the cherry-pick lines need to be in a separate comment, not the PR description.

**Comment by phoneben** [comment]
cherry-pick-to: 20
cherry-pick-to: 18

**Comment by phoneben** [comment]
get to messy will open a new PR the corect way could not figure how to squash the commits

**Pull request closed without merging**